### PR TITLE
feat(api): remove csrf protection from get-session-user

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

get-session-user doesn't mutate anything, so it should be safe from a CSRF perspective.

### CODE REVIEW

**Comment by naomi-lgbt** [comment]
> so it should be safe from a CSRF perspective.

Admittedly a small attack vector, but should we not be concerned about a CSRF attack exposing the camper's private information (mainly the email, but also anything they might have marked as private in their profile)?

**Comment by ojeytonwilliams** [comment]
I don't believe that's possible, even with these changes. The attacks that CSRF protections defend against are ones where a legit user visits a site under the attacker's control. If that site makes a request to _our_ site, then that request will include the user's credentials (in cookie form). The attacker could make a malicious request (i.e. POST to /account/delete). Since the user's credentials are included, that could go through.

However, the attacker never receives the response, because we set `Access-Control-Allow-Origin` to urls that we control. So, even though the data is transmitted, the attacker can't access it - the browser doesn't allow it.

Basically, CSRF attacks are all about the side-effects and get-session-user has none.

**Review by naomi-lgbt — APPROVED** [review]
Okie dokie

**Comment by raisedadead** [comment]
>  because we set Access-Control-Allow-Origin to urls that we control.

This will change when we allow apps like Classroom to be self-hosted (a primary use-case for the app) on-premises and still interact with fCC API.

**Comment by ojeytonwilliams** [comment]
That's a post-MVP issue, no? 

For the MVP I'm focused on making the new api compatible with the web and mobile apps.

**Review by moT01 — APPROVED** [review]
This LGTM if we want it 🎉 

I guess my question is why we wouldn't keep this on all endpoints that don't explicitly need it removed?

**Comment by moT01** [comment]
Going to merge this so I can test that other PR - this is all on the new API so it's not going into production yet. We can revisit this conversation if/when needed.

**Comment by ojeytonwilliams** [comment]
> I guess my question is why we wouldn't keep this on all endpoints that don't explicitly need it removed?

In general we should keep it, but the client can end up making this request before it has the token. We can definitely revisit this post-MVP, though.
